src/python/pants/util/ordered_set.py

```diff
@@ -222,37 +161,20 @@ def symmetric_difference(self, *others: Iterable[T]) -> "OrderedSet[T]":
 
     def add(self, key: T) -> None:
         """Add `key` as an item to this OrderedSet."""
-        if key in self:
-            return
-        self._items.append(key)
-
-    append = add
+        self._items[key] = None
 
     def update(self, iterable: Iterable[T]) -> None:
         """Update the set with the given iterable sequence."""
         for item in iterable:
             self.add(item)
 
-    extend = update
-
-    def pop(self) -> T:
-        """Remove and return the last element from the set.
-
-        Raises KeyError if the set is empty.
-        """
-        if not self._items:
-            raise KeyError("OrderedSet is empty")
-        return self._items.pop()
```

Comment:
Is the idea that we inherit `MutableSet`'s `pop()` implementation? https://docs.python.org/3/library/collections.abc.html

Reply:
Right: the primary difference between `set` pop and `Sequence` pop was that the former pops something random.